logic/broadcaster.js

```diff
@@ -242,21 +247,21 @@ __private.filterQueue = function(cb) {
 };
 
 /**
- * Checks if transaction is in pool or confirm it.
+ * Checks if transaction of a given id is in pool or confirmed in database "trs" table.
  *
  * @private
- * @param {transaction} transaction
+ * @param {string} transactionId
  * @param {function} cb
  * @returns {SetImmediate} null, boolean
  * @todo Add description for the params
  */
-__private.filterTransaction = function(transaction, cb) {
-	if (transaction !== undefined) {
-		if (modules.transactions.transactionInPool(transaction.id)) {
+__private.filterTransaction = function(transactionId, cb) {
```

Comment:
The function name doesn't do what it says, it's not filtering the transaction rather it just checks if the transaction exists in transaction pool or transaction exists in the database, so it would be good to rename it to something meaningful like `isTransactionExistsInPoolOrDatabase`

Reply:
Agree to Manu to rename this function to something `isTransactionExistsInPoolOrDatabase`

Reply:
Not really in the scope of the bug fix I hope, but I agree, I will rename on 2 corresponding functions.